src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -117,34 +117,46 @@ impl CompactionPicker for MinOverlappingPicker {
         &mut self,
         levels: &Levels,
         level_handlers: &[LevelHandler],
-        stats: &mut LocalPickerStatistic,
-    ) -> Option<CompactionInput> {
+    ) -> (Option<CompactionInput>, Vec<LocalPickerStatistic>) {
         assert!(self.level > 0);
+        let mut local_picker_stats = Vec::default();
+        let mut min_overlapping_picker_stat = LocalPickerStatistic::new(
+            self.level,
+            self.target_level,
+            format!("{} -> {} ", self.level, self.target_level),
+        );
         let (select_input_ssts, target_input_ssts) = self.pick_tables(
             &levels.get_level(self.level).table_infos,
             &levels.get_level(self.target_level).table_infos,
             level_handlers,
         );
+
         if select_input_ssts.is_empty() {
-            stats.skip_by_pending_files += 1;
-            return None;
+            min_overlapping_picker_stat.set_skip_by_pending_files();
+            local_picker_stats.push(min_overlapping_picker_stat);
+            return (None, local_picker_stats);
         }
-        Some(CompactionInput {
-            input_levels: vec![
-                InputLevel {
-                    level_idx: self.level as u32,
-                    level_type: LevelType::Nonoverlapping as i32,
-                    table_infos: select_input_ssts,
-                },
-                InputLevel {
-                    level_idx: self.target_level as u32,
-                    level_type: LevelType::Nonoverlapping as i32,
-                    table_infos: target_input_ssts,
-                },
-            ],
-            target_level: self.target_level,
-            target_sub_level_id: 0,
-        })
+
+        local_picker_stats.push(min_overlapping_picker_stat);
```

Comment:
There is no skip happening if we reach here. I don't get why we need to push the picker_stat here.